enumerable.rb

```diff
@@ -0,0 +1,213 @@
+module Enumerable
+  def my_each
+    i = 0
+    if block_given?
+      while i < to_a.length
+        yield to_a[i]
+        i += 1
+      end
+      self
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    if block_given?
+      while i < to_a.length
+        yield(to_a[i], i)
+        i += 1
+      end
+      self
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    new_items = []
+    if block_given?
+      to_a.my_each { |item| new_items.push(item) if yield item }
+      new_items
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(param = nil)
+    my_all_flag = true
+    if block_given?
+      to_a.my_each do |item|
+        return my_all_flag = false if item.nil? || item == false
+        return my_all_flag = false unless yield item
+      end
+    elsif param
+      if param.instance_of?(Class)
+        my_each do |item|
+          return my_all_flag = false if item.is_a?(param) == false
+        end
+      elsif param.instance_of?(Regexp)
+        my_each do |item|
+          return my_all_flag = false if param.match(item).nil?
+        end
+        # elsif !param.is_a?(Class) && !param.is_a?(Regexp)
+        #   case param
+        #     in self
+        #       return true
+        #   else
+        #     return false
+        #   end
+      end
+    else
+      to_a.my_each do |item|
+        return my_all_flag = false if item.nil? || item == false
+      end
+    end
+    my_all_flag
+  end
+
+  def my_any?(param = nil)
```

Comment:
- [ ] **enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true**

Case
```
false_array.my_any?
```
```
expected false
            got true
```